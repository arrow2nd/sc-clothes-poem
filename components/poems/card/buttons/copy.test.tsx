import { act, render } from '@testing-library/react'

import CopyButton from './copy'

describe('CopyButton', () => {
  beforeEach(() => {
    // 未実装エラーの対策
    window.prompt = jest.fn()
  })

  test('スナップショットテスト', () => {
    const { container } = render(<CopyButton text="asahi" />)
    expect(container).toMatchSnapshot()
  })

  test('クリックするとアイコンが変わるかどうか', () => {
    const { container } = render(<CopyButton text="mei" />)

    act(() => {
      container.children[0].dispatchEvent(
        new MouseEvent('click', { bubbles: true })
      )
    })

    expect(container).toMatchSnapshot()
  })
})
