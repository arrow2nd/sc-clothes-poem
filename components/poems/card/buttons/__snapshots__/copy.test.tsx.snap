// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton クリックするとアイコンが変わるかどうか 1`] = `
<div>
  <button
    class="text-xl hover:text-black transition-colors focus:outline-none"
    title="コピーする"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
      />
    </svg>
  </button>
</div>
`;

exports[`CopyButton スナップショットテスト 1`] = `
<div>
  <button
    class="text-xl hover:text-black transition-colors focus:outline-none"
    title="コピーする"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="13"
        rx="2"
        ry="2"
        width="13"
        x="9"
        y="9"
      />
      <path
        d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
      />
    </svg>
  </button>
</div>
`;
